python/dgl/data/utils.py

```diff
@@ -482,3 +484,198 @@ def add_nodepred_split(dataset, ratio, ntype=None):
         g.nodes[ntype].data["train_mask"] = train_mask
         g.nodes[ntype].data["val_mask"] = val_mask
         g.nodes[ntype].data["test_mask"] = test_mask
+
+
+def mask_nodes_by_property(property_values, part_ratios, random_seed=None):
```

Comment:
Do we need to expose this function to users?

Reply:
Yes, I think this function should be exposed to users, as some of them may already know which node property they want to use, so they just need to split the given data and induce their custom distributional shift. I have added an example in the doc string and the corresponding unit test function.